app/setup.php

```diff
@@ -120,9 +120,9 @@
     });
 
     /**
-     * Create @asset() Blade directive
+     * Register Directives with Blade
      */
-    sage('blade')->compiler()->directive('asset', function ($asset) {
-        return "<?= " . __NAMESPACE__ . "\\asset_path({$asset}); ?>";
+    collect(config('directives'))->each(function ($directive, $function) {
+        sage('blade')->compiler()->directive($function, $directive);
```

Comment:
OK, this is working (👍), but I think the variable are mismatched.
I mean, currently we have:
```
$directive: Closure (the function)
$function: String (the directive name)
```
and it seems more intuitive to have
```
$directive: String (the directive name)
$function: Closure (the function)
```

Right? 🙂 

(using [Laravel names](https://github.com/laravel/framework/blob/19080684fb06fd09cd0882e3b19f7847c660cec1/src/Illuminate/View/Compilers/BladeCompiler.php#L467-L477) could also be relevant here)

Reply:
I can dig it. I switched to the Laravel names as well as swapped the order. The order was a mistake when I changed from `array_map()` to `->each()` which outputs the vars oppositely.